location-search/SearchClient.ts

```diff
@@ -0,0 +1,148 @@
+  query: LocationsSearchQuery,
+  center?: LocationCoordinate2D
+) => {
+  const options = {
+    maxResults: 10,
+    biasPosition: center ? [center?.longitude, center?.latitude] : undefined
+  }
+  const geoSearchResults = await Geo.searchByText(query.toString(), options)
+  return ArrayUtils.compactMap(geoSearchResults, awsPlaceToTifLocation)
+}
+
+/**
+ * A function that pulls the 10 most recent locations stored in AsyncStorage, and maps out their data
+ * into a set of {@link LocationSearchResult}s.
+ * @returns A Promise of an array of {@link LocationSearchResult}s.
+ */
+export const searchRecentLocations = (): Promise<LocationSearchResult[]> => {
+  return asyncStorageLoadRecentLocations(10).then((results) =>
+    results.map((area) => ({ ...area, isRecentLocation: true }))
+  )
+}
+
+/**
+ * A function that uses a provided query, with both its provided searchFunction to obtain {@link TiFLocation}s,
+ * and {@link asyncStorageLoadSpecificRecentLocations} on their coordinates to obtain async storage's information
+ * on whether those locations are recent.
+ * This involves combining the data from both processes.
+ *
+ * @param query A {@link LocationsSearchQuery} given for the search function to use.
+ * @param searchFunction A function that takes in a query + an optional center, then converts the data into a usable set of {@link TiFLocation}s.
+ * @param center An optional {@link LocationCoordinate2D} given for the search to center in where it should be searching.
+ * @returns A Promise of an array of {@link LocationSearchResult}s.
+ */
+
+export const searchWithRecentAnnotations = async (
+  query: LocationsSearchQuery,
+  searchFunction: (
+    query: LocationsSearchQuery,
+    center?: LocationCoordinate2D
+  ) => Promise<TiFLocation[]>,
+  center?: LocationCoordinate2D
+): Promise<LocationSearchResult[]> => {
+  const remoteSearchResults = await searchFunction(query, center)
+  const convertedRemoteSearchResults = remoteSearchResults.map((point) => ({
+    location: point
+  }))
+  console.log(
+    "Converted Remote Search: " + JSON.stringify(convertedRemoteSearchResults)
+  )
+  const searchCoordinates = remoteSearchResults.map(
+    (point) => point.coordinates
+  ) as LocationCoordinate2D[]
+  const asyncRecentSearchResults =
+    await asyncStorageLoadSpecificRecentLocations(searchCoordinates)
+  const filteredResults = ArrayUtils.compactMap(
+    convertedRemoteSearchResults,
+    (remoteSearchPoint) => {
+      if (
+        !asyncRecentSearchResults.find(
+          (asyncPoint) =>
+            JSON.stringify(asyncPoint.location) ===
+            JSON.stringify(remoteSearchPoint.location)
+        )
+      ) {
+        return remoteSearchPoint
+      }
+    }
+  )
+  const mergedResults = asyncRecentSearchResults.concat(filteredResults)
+  return mergedResults.map((result) => ({
+    location: result.location,
+    annotation: result.annotation,
+    isRecentLocation: !!result.annotation
+  }))
```

Comment:
I think what I would do here is map on the `remoteSearchResults`, then do the `.find` stuff you did earlier. This would preserve the ordering whilst still merging the data properly.